epictrack-web/src/components/workPlan/event/components/ExtensionInput.tsx

```diff
@@ -0,0 +1,173 @@
+import { Grid, TextField } from "@mui/material";
+import React from "react";
+import Moment from "moment";
+import { AdapterDayjs } from "@mui/x-date-pickers/AdapterDayjs";
+import dayjs from "dayjs";
+import { ETFormLabel, ETFormLabelWithCharacterLimit } from "../../../shared";
+import ControlledSelectV2 from "../../../shared/controlledInputComponents/ControlledSelectV2";
+import { Controller, useFormContext } from "react-hook-form";
+import { ListType } from "../../../../models/code";
+import { DatePicker, LocalizationProvider } from "@mui/x-date-pickers";
+import { DATE_FORMAT } from "../../../../constants/application-constant";
+import { WorkplanContext } from "../../WorkPlanContext";
+import { dateUtils } from "../../../../utils";
+import { SyntheticEvent } from "react-draft-wysiwyg";
+
+interface ExtensionInputProps {
+  isFormFieldsLocked: boolean;
+}
+const ExtensionInput = (props: ExtensionInputProps) => {
+  const [actSections, setActSections] = React.useState<ListType[]>([]);
+  const [reasonCount, setReasonCount] = React.useState<number>(0);
+  const changeReasonTextHandler = (event: any) => {
+    setReasonCount(event.target.value.length);
+  };
+  const {
+    register,
+    unregister,
+    formState: { errors },
+    control,
+    setValue,
+    getValues,
+  } = useFormContext();
+  const ctx = React.useContext(WorkplanContext);

```

Comment:
What's your opinion about destructuring, insterad of having ctx and always accessing the props by doing ctx. we destructure here const { selectedWorkPhase } = useContext...